### PR TITLE
Allow running tests on Windows by using `SWIFTSYNTAX_BUILD_DYNAMIC_LIBRARY`

### PR DESCRIPTION
**Pull request by ahoppen**
Also, fix two minor Windows test failures.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

File: Package.swift [hunk 1]
```diff
@@ -460,6 +458,13 @@ var installAction: Bool { hasEnvironmentVariable("SOURCEKIT_LSP_CI_INSTALL") }
 /// remote dependency.
 var useLocalDependencies: Bool { hasEnvironmentVariable("SWIFTCI_USE_LOCAL_DEPS") }
 
+/// Whether swift-syntax is being built as a single dynamic library instead of as a separate library per module.
+///
+/// This means that the swift-syntax symbols don't need to be statically linked, which alles us to stay below the
```
Comment (bnbarham) [inline Package.swift:463]:
```suggestion
/// This means that the swift-syntax symbols don't need to be statically linked, which allows us to stay below the
```

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**